includes/tracking-events.php

```diff
@@ -1,120 +1,87 @@
-                . "');"
-        
+                "window.metrilo.search('$search_query', '$search_link');"
             );
-            $this->single_item_tracked = true;
         }
         
         // if visitor is viewing shopping cart page
-        if(!$this->single_item_tracked && is_cart()){
+        if(is_cart()){
             $this->put_event_in_queue(
                 "window.metrilo.customEvent('view_cart');"
             );
-            $this->single_item_tracked = true;
         }
         // if visitor is anywhere in the checkout process
-        if (!$this->single_item_tracked && is_order_received_page()) {
+        if (is_order_received_page()) {
+            $url  = get_permalink();
+            $name = json_encode(['name' => get_the_title()]);
             $this->put_event_in_queue(
-                "window.metrilo.viewPage('"
-                . get_permalink()
-                . "', "
-                . json_encode(['name' => get_the_title()])
-                . ");"
+                "window.metrilo.viewPage('$url', $name);"
             );
-            $this->single_item_tracked = true;
-            
-        } elseif (
-            !$this->single_item_tracked &&
-            function_exists('is_checkout_pay_page') &&
-            is_checkout_pay_page()
-        ) {
+        } elseif (function_exists('is_checkout_pay_page') && is_checkout_pay_page()) {
             $this->put_event_in_queue(
                 "window.metrilo.customEvent('checkout_payment');"
             );
-            $this->single_item_tracked = true;
-        } elseif (!$this->single_item_tracked && is_checkout()) {
+        } elseif (is_checkout()) {
             $this->put_event_in_queue(
                 "window.metrilo.checkout();"
             );
-            $this->single_item_tracked = true;
         }
     }
     //
     //  ** GENERIC WordPress tracking - doesn't require WooCommerce in order to work **//
     //
     //  if visitor is viewing homepage or any text page
-    if (!$this->single_item_tracked && is_front_page()) {
+    if (is_front_page()) {
+        $url  = get_permalink();
+        $name = json_encode(['name' => get_the_title()]);
         $this->put_event_in_queue(
-            "window.metrilo.viewPage('"
-            . get_permalink()
-            . "', "
-            . json_encode(['name' => get_the_title()])
-            . ");"
+            "window.metrilo.viewPage('$url', $name);"
         );
-        $this->single_item_tracked = true;
-    } elseif (!$this->single_item_tracked && is_page()) {
+    } elseif (is_page() && !is_cart() && !is_checkout()) {
+        $url  = get_permalink();
+        $name = json_encode(['name' => get_the_title()]);
         $this->put_event_in_queue(
-            "window.metrilo.viewPage('"
-            . get_permalink()
-            . "', "
-            . json_encode(['name' => get_the_title()])
-            . ");"
+            "window.metrilo.viewPage('$url', $name);"
         );
```

Comment:
This `if` bodies are still the same. The `if (is_order_received_page()) { ... }` body also.